components/candle/masteryService.ts

```diff
@@ -80,10 +83,9 @@ export class MasteryService {
         gameVersion: GameVersion,
         userId: string,
     ): MasteryDataTemplate {
-        const subLocation: Unlockable = getSubLocationByName(
-            locationId,
-            gameVersion,
-        )
+        const location: Unlockable =
+            getSubLocationByName(locationId, gameVersion) ??
+            getParentLocationByName(locationId, gameVersion)
```

Comment:
I have mixed feelings about this code. I feel we have multiple ways of achieving the same result now. Do you actually need the `template` that is part of this response, or are you only interested in the `MasteryData` - in that case, you should just call getMasteryData directly.

Can we check if all usages of these two functions is consistent now over the codebase of Peacock?

Reply:
This function is only used by `MasteryDataForLocation`, the Career -> Mastery page requests this endpoint _by parent location_ thus is the reason for adding parent location handling to this function.